source/extensions/filters/network/dubbo_proxy/metadata.h

```diff
@@ -0,0 +1,128 @@
+
+#pragma once
+
+#include <map>
+#include <string>
+
+#include "common/http/header_map_impl.h"
+
+#include "extensions/filters/network/dubbo_proxy/protocol.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace DubboProxy {
+
+class MessageMetadata {
+public:
+  struct ParameterValue {
+    ParameterValue(uint32_t index, const std::string& value) : index_(index), value_(value) {}
```

Comment:
do you need this? You can do `ParameterValue{index, value}` without the constructor.

Actually do you need this struct? Seems index always is the key of the map? So why not just an `unorderd_map<uint32_t, std::string>`?

Reply:
Initially, this structure had not only two values, index and value, but also a type that represented the parameter type. however, since there is currently no implementation of the data type mapping, i have removed it, kept the structure for future use, and added a TODO entry to myself

Reply:
I still don't think you need index inside the value, what's the point to have that? And no need of the constructor either.

Reply:
ok, i have deleted ParameterValue structure